haystack/nodes/reader/base.py

```diff
@@ -77,10 +77,11 @@ def add_doc_meta_data_to_answer(documents: List[Document], answer):
             answer.meta = {}
         # get meta from doc
         meta_from_doc = {}
-        for doc in documents:
-            if doc.id == answer.document_id:
-                meta_from_doc = deepcopy(doc.meta)
-                break
+        if answer.document_ids:
+            for doc in documents:
+                if doc.id in answer.document_ids:
+                    meta_from_doc = deepcopy(doc.meta)
+                    break
```

Comment:
This is the Reader code that copies over the document meta to Answer's meta. It had to be adjusted for the type change. Note that `len(answer.document_ids)` can only be 1, 0 or `answer.document_ids` is `None`.